\name{pscn.plot}
\alias{pscn.plot}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{Plot function used to visualize parent specific copy number estimation (pscn) results.}
\description{
This function provides a variety of plots showing results of a pscn analysis after \code{\link{smoothing}}, \code{\link{segmentation}} and \code{\link{pscnlist}}.   
}
\usage{
pscn.plot(cnvobj = myobj, ...)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{cnvobj}{Need to be specified when which.plot is "bfreq", "logR", "A", or "B". It is the result from \code{\link{segmentation}}, by loading .pscn.Rdata file.}
  \item{which.plot}{can only be any of "bfreq" (plot Bfreqency), "logR" (plot logR), "A" (plot allele A intensity), "B" (plot allele B intensity), "copy number" (plot major and minor copy numbers). (Default="bfreq").}
  \item{region}{Applicable when which.plot is "bfreq", "logR", "A", or "B". If TRUE, plot different segments with different colors; otherwise, plot the four states with different colors.  (Default=FALSE)}
  \item{regionid}{Applicable when which.plot is "bfreq", "logR", "A", or "B". Specifies the segments to be ploted.}
  \item{region.col}{Applicable when which.plot is "bfreq", "logR", "A", or "B". Specifies the colors used for segments. }
  \item{scatter}{If TRUE, plots allele B intersity vs. allele A intensity.  (Default=FALSE)}
  \item{Contour}{create contour plot for B vs A.  (Default=FALSE)}
  \item{use.main}{If TRUE, main title will be generated.  (Default=FALSE)}
  \item{cndata}{the shortlist generated by the preceeding step \link{pscnlist}. Have to be specified when which.plot is "copy number".}
  \item{chrid}{Need to be specified when which.plot is "copy number". Any one value between 1 and 23. Indicates the chromosome to be ploted.  If NULL (default), all chromosome(s) will be plotted.}
  \item{use.pos}{If TRUE, the x axis will be position (bp) rather than SNP index.  (Default=FALSE)}
  \item{loc}{specifies the SNPs to be ploted, eg 2000:3000.   If NULL (default), all SNPs will be plotted.}
  \item{changepoint}{Applicable when which.plot is "bfreq", "logR", "A", or "B". If TRUE, changepoints will be indicated by black vertical lines. (Default = TRUE)}
  \item{col.gain}{Applicable when which.plot is "copy number", to specifies the color for gain chromosome regions.  (Default=red)}
  \item{col.loss}{Applicable when which.plot is "copy number", to specifies the color for loss chromosoem regions.  (Default=blue)}
  \item{col.normal}{Applicable when which.plot is "copy number", to specifies the color for normal chromosome regions.  (Default=green)}
}
\references{ 
Chen, H., Xing, H. and Zhang, N.R. (2009) Estimation of parent specific DNA copy number
in tumors using high-density genotyping arrays.  Under Review, manuscript can be downloaded from www-stat.stanford.edu/~nzhang/.
 }
\author{Hao Chen}
\note{All plots can only be done after calling the function \link{pscnlist}.}
\seealso{
This function provides a variety of plots showing results after the pscn analysis after \link{smoothing}, \link{segmentation} and \link{pscnlist}.   
}
\examples{

# First load the example data set, call smoothing and segmentation functions, then call pscnlist to generate analysis results.
  data(ex0) # this file contain four columns "Chromosome ID", "SNP Position (bp)", "logR" and "Bfrequency" with column names "Chr", "Position", "logR" and "bfreq".

# smoothing (estimate parent-specify copy numbers)
  smoothing(samplename="TCGA.02.0043.01A.01D", inputdata=mydata, platform="Illumina") 

# segmentation (determine change points) this function can only be run after the function "smoothing".
  segmentation(samplename="TCGA.02.0043.01A.01D",chrs=1,regroup.percent=0.05) 

# pscnlist (get a table showing major and minor copy numbers) this function can only be run after the function "segmentation".
  pscnlist(samplename="TCGA.02.0043.01A.01D", chrs=1)

# view major and minor copy numbers of a chromosome (use result from \link{pscnlist}).

  mydata = read.table("TCGA.02.0043.01A.01D.shortlist.txt", header=TRUE)    # read in a Text document named samplename.shortlist.txt for "copy number" plot 
  pscn.plot(cndata=mydata, which.plot="copy number", chrid=1)
  # if want x axis to be position rather than SNP index and show title
  pscn.plot(cndata=mydata, which.plot="copy number", chrid=1, use.pos=TRUE, use.main=TRUE) # the disconneted region indicates cetromere region
  # change gain, loss, normal colors  
  pscn.plot(cndata=mydata, which.plot="copy number", chrid=1, use.pos=TRUE, use.main=TRUE, col.gain="purple", col.loss="yellow", col.normal="lightblue")
  
# view logR, Bfreqency, allele A intensity or allele B intensity (use result from \link{segmentation} and \link{pscnlist}) )
  load("TCGA.02.0043.01A.01D.Chr1.pscn.Rdata")  # load samplename.Chr1.pscn.Rdata  (Chr1 can be replaced by any other chromosomes, such as Chr2, etc.) for plot other than "copy number".
  mydata = read.table("TCGA.02.0043.01A.01D.shortlist.txt", header=TRUE)    # read in a Text document named samplename.shortlist.txt for "copy number" plot
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq") # change which.plot to logR, A or B to view logR, allele A intensity or allele B intensity, same following.
  # if do not want to see the black vertical lines showing changepoint
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", changepoint=FALSE)
  # to color different segment differently
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", region=TRUE)
  # only view SNPs from 2000 to 5000 
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", loc=2000:5000)   
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", changepoint=FALSE, loc=2000:5000)        
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", region=TRUE, loc=2000:5000)
  # only view segments 4-8                                               
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", region=TRUE, regionid=4:8)
  pscn.plot(cnvobj, cndata=mydata, which.plot="bfreq", region=TRUE, regionid=4:8, region.col=c("blue","green","yellow","red","black")) #specifies the colors for segments by oneself
  
# scatter plot
  load("TCGA.02.0043.01A.01D.Chr1.pscn.Rdata")  # load samplename.Chr1.pscn.Rdata  (Chr1 can be replaced by any other chromosomes, such as Chr2, etc.) for plot other than "copy number".
  mydata = read.table("TCGA.02.0043.01A.01D.shortlist.txt", header=TRUE)    # read in a Text document named samplename.shortlist.txt for "copy number" plot
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE)  # color the four states differently
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, region=TRUE)  # color different segment differently
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, loc=2000:5000) # only view SNPs from 2000 to 5000
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, region=TRUE, loc=2000:5000)
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, regionid=2)
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, region=TRUE, regionid=2:3)
  pscn.plot(cnvobj, cndata=mydata, scatter=TRUE, region=TRUE, regionid=2:3, region.col=c("blue","green"))

# contour plot
  load("TCGA.02.0043.01A.01D.Chr1.pscn.Rdata")  # load samplename.Chr1.pscn.Rdata  (Chr1 can be replaced by any other chromosomes, such as Chr2, etc.) for plot other than "copy number".
  mydata = read.table("TCGA.02.0043.01A.01D.shortlist.txt", header=TRUE)    # read in a Text document named samplename.shortlist.txt for "copy number" plot
  library(MASS)
  pscn.plot(cnvobj, cndata=mydata, Contour=TRUE)  # color different segment differently
  pscn.plot(cnvobj, cndata=mydata, Contour=TRUE, loc=2000:5000) # only view SNPs from 2000 to 5000
  pscn.plot(cnvobj, cndata=mydata, Contour=TRUE, regionid=2)
  pscn.plot(cnvobj, cndata=mydata, Contour=TRUE, regionid=2:3)
  pscn.plot(cnvobj, cndata=mydata, Contour=TRUE, region=TRUE, regionid=2:3, region.col=c("blue","green"))
}
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
